Tools/TestWebKitAPI/Tests/WebKitCocoa/RestoreLocalStorage.mm

```diff
@@ -119,8 +174,46 @@ async function setItem()
 </script>
 )TESTRESOURCE"_s;
 
-static void testFetchLocalStorageThirdPartyIFrame(RetainPtr<WKWebsiteDataStore> websiteDataStore)
+static NSString *getMainFrameString = @"<script> \
+    function postResult(event) \
+    { \
+        window.webkit.messageHandlers.testHandler.postMessage(event.data); \
+    } \
+    addEventListener('message', postResult, false); \
+    </script> \
+    <iframe src='https://127.0.0.1:9091/'>";
+
+static constexpr auto getFrameBytes = R"TESTRESOURCE(
+<script>
+function postMessage(message)
 {
+    parent.postMessage(message, '*');
+}
+async function getItem()
+{
+    try {
+        postMessage(localStorage.getItem('key'));
+    } catch(err) {
+        postMessage('error: ' + err.name + ' - ' + err.message);
+    }
+}
```

Comment:
Not quite sure why this function is `async` -- `getItem` is sync; and exception is unlikely to throw with existing setup (so `try... catch` is also unnecessary).